certora/specs/methods/IERC3156.spec

```diff
@@ -0,0 +1,5 @@
+methods {
+    maxFlashLoan(address)                    returns (uint256) envfree => DISPATCHER(true)
+    flashFee(address,uint256)                returns (uint256) envfree => DISPATCHER(true)
+    flashLoan(address,address,uint256,bytes) returns (bool)            => DISPATCHER(true)
+}
```

Comment:
What exactly does it mean to include this file? I had a look at the [`methods`](https://docs.certora.com/en/latest/docs/cvl/methods.html) docs but could get a good intuition.

Reply:
AFAIK, declaring the methods is optional, but it can bring some added values

- One of the value is that if you use one of this methods in a spec file, but the contract targeted doesn't implement them, then the specs checking will not fail, it will just skip the tests.
  - That is something we can see in ERC20.spec. [This run](https://prover.certora.com/output/71959/2513b40ab3864c428cb5e3ed44a504cb/?anonymousKey=c8e9bd5b655e025c994c5182bcc41be711444c64) shows what happen it the contract doesn't publicly expose mint and burn, the tests are skipped and the rest of the property is verified
  - IMO this is great for test reuse
  
- Another value, is that it allows us to specify that the method is envfree (the whole concept of envfree is very strange to me. A function that reads from storage can be envfree, but not one that reads block.timestamp) 

- Lastly it enable "summaries" ... Now that is a concept I find very hard to understand, but certora that `DISPATCHER(true)` should be used for "standard interfaces such as ERC20" ... so I follow their guideline.